From 55904d335d88e2126545bb9660bb8429b8909c57 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <karel.koci@nic.cz>
Date: Thu, 16 May 2019 15:52:45 +0200
Subject: [PATCH] base-files: run switch-brach --verify after update

When base-files is updated it replaces its distfeeds to default. That
moves it from selected branch back to deploy. We are unable to do that
change only on server because some branches are just binary copies of
less stable branch. This solves it just by setting current branch back
in distfeeds.
---
 package/base-files/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/package/base-files/Makefile b/package/base-files/Makefile
index c16091b6bc..dd57ebb297 100644
--- a/package/base-files/Makefile
+++ b/package/base-files/Makefile
@@ -37,7 +37,7 @@ endif
 define Package/base-files
   SECTION:=base
   CATEGORY:=Base system
-  DEPENDS:=+netifd +libc +procd +jsonfilter +SIGNED_PACKAGES:usign +SIGNED_PACKAGES:openwrt-keyring +NAND_SUPPORT:ubi-utils +fstools +fwtool
+  DEPENDS:=+netifd +libc +procd +jsonfilter +SIGNED_PACKAGES:usign +SIGNED_PACKAGES:openwrt-keyring +NAND_SUPPORT:ubi-utils +fstools +fwtool +switch-branch
   TITLE:=Base filesystem for OpenWrt
   URL:=http://openwrt.org/
   VERSION:=$(PKG_RELEASE)-$(REVISION)
@@ -208,6 +208,8 @@ define Package/base-files/postinst
 			"/etc/init.d/$$L" enable
 		fi
 	done < /etc/services_wanted
+	# Update new /etc/opkg/distfeeds.conf to point to correct repository
+	switch-branch --verify
 }
 endef
 
-- 
2.21.0

